OpenRA.Mods.Common/Scripting/Properties/ProductionProperties.cs

```diff
@@ -46,7 +46,8 @@ public void Produce(string actorType, string factionVariant = null, string produ
 				// Go through all available traits and see which one successfully produces
 				foreach (var p in productionTraits)
 				{
-					if (!string.IsNullOrEmpty(productionType) && !p.Info.Produces.Contains(productionType))
+					var type = productionType ?? actorInfo.TraitInfo<BuildableInfo>().BuildAtProductionType;
```

Comment:
The TraitInfo query can move outside the loop to save unnecessary lookups if there are multiple production traits.

Reply:
Derp. Fixed.